recipe/build.sh

```diff
@@ -1,24 +1,27 @@
 #!/bin/bash
 
 if [[ $HOST == *apple* ]]; then
-    # Apply patch to build.py.in (autoreconf might not be needed?)
-    patch -p1 < ${RECIPE_DIR}/osx_patch.diff
-    # These two flags are mentioned in the LHAPDF docs
+    # Bug for macOS as of lhapdf v6.1.1+
+    # https://gitlab.com/hepcedar/lhapdf/-/blob/c5b048f5794e483d85801d1ae39f42aaa46999dd/INSTALL#L162-164
     export CFLAGS=-Qunused-arguments
     export CPPFLAGS=-Qunused-arguments
-fi
 
-if [[ $HOST == *arm64* ]]; then
-    # Get an updated config.sub and config.guess
-    cp $BUILD_PREFIX/share/gnuconfig/config.* ./config/
-    # And autoreconf!
-    autoreconf -i -f
-    (cd wrappers/python && cython --cplus lhapdf.pyx)
+    # As using autotools
+    # https://conda-forge.org/blog/2020/10/29/macos-arm64/#how-to-add-a-osx-arm64-build-to-a-feedstock
+    if [[ $HOST == *arm64-apple* ]]; then
+        # Get an updated config.sub and config.guess
+        cp $BUILD_PREFIX/share/gnuconfig/config.* .
+    fi
 fi
 
-./configure --prefix=$PREFIX
+autoreconf --install --force
+(cd wrappers/python && cython --include-dir . --cplus lhapdf.pyx)
```

Comment:
Generate the Python sources with Cython for all build platforms given @chrisburr's recommendation that this can be useful in cases like new Python versions.